projects/optic/src/commands/capture/capture.ts

```diff
@@ -127,58 +134,56 @@ const getCaptureAction =
     } else {
       chdir(captureConfig.server.dir);
     }
+    const serverUrl = options.serverOverride || captureConfig.server.url;
     let sourcesController = new AbortController();
     let [proxyInteractions, proxyUrl] = await ProxyInteractions.create(
-      captureConfig.server.url,
+      serverUrl,
       sourcesController.signal,
       {
         mode: 'reverse-proxy',
         proxyPort: options.proxyPort ? Number(options.proxyPort) : undefined,
       }
     );
 
-    // start app
-    const cmd = captureConfig.server.command.split(' ')[0];
-    const args = captureConfig.server.command.split(' ').slice(1);
-    const app = spawn(cmd, args, { detached: true });
+    // start the app
+    let app: ChildProcessWithoutNullStreams;
+    if (!options.serverOverride && captureConfig.server.command) {
+      const cmd = captureConfig.server.command.split(' ')[0];
+      const args = captureConfig.server.command.split(' ').slice(1);
+      app = spawn(cmd, args, { detached: true });
 
-    // log error output from the app
-    app.stderr.on('data', (data) => {
-      console.log(data.toString());
-    });
+      // log error output from the app
+      app.stderr.on('data', (data) => {
+        console.log(data.toString());
+      });
+    }
 
     // wait until server is ready
-    await new Promise((r) =>
-      setTimeout(r, captureConfig.server.ready_interval)
-    );
-
-    if (captureConfig.server.ready_endpoint) {
```

Comment:
i struggle to think that theres a situation where we don't want to wait until the server is ready

Reply:
the point about the root endpoint not being available is valid.

as far as not needing to check for an external server—consider my demo. in that example our "external" server is not ready (because i hadnt started it before starting capture). the ready check gets us the pleasant behavior of not needing to have the proxy and the server be ready in a strict order.

that said, i can certainly imagine situations where failing faster is preferable. so yeah, i suppose you're right here. we'll only wait if a ready_endpoint is supplied.

Reply:
so here's the thing: we _have_ to have some sort of wait to handle the local server case. using my go test server, which will start about as fast as anything we can expect, capture sends the requests before the server is ready. we weren't seeing that before because we had that one superfluous seeming timeout right before dropping into the loop.  so i think we need to either:

1. keep the arbitrary sleep of one internal before we start server checking
2. or, require ready_endpoint and always ready check

Reply:
going with #1